### ISSUE TITLE
Cant download any package on Windows

### ISSUE THREAD

**Issue by Parka** [runk/npm-proxy-cache#26]
It runs ok, but upon doing "npm install" I get the following:
```
npm ERR! Windows_NT 10.0.10586
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install"
npm ERR! node v4.2.4
npm ERR! npm  v2.14.12
npm ERR! code ECONNRESET

npm ERR! network tunneling socket could not be established, statusCode=500
npm ERR! network This is most likely not a problem with npm itself
npm ERR! network and is related to network connectivity.
npm ERR! network In most cases you are behind a proxy or have bad network settings.
npm ERR! network
npm ERR! network If you are behind a proxy, please make sure that the
npm ERR! network 'proxy' config is set properly.  See: 'npm help config'

npm ERR! Please include the following file with any support request:
npm ERR!     C:\Users\parka\Documents\projects\localum\phaser-template\generator-phaser-template-master\npm-debug.log
```

And then on the proxy window:
```
[2016-01-17 21:36:06.581] [INFO] proxy - Listening on localhost:8080 [10204]
[2016-01-17 21:37:09.460] [DEBUG] proxy -   = will connect to socket "C:\Users\parka\AppData\Local\Temp\mitm.sock"
[2016-01-17 21:37:09.460] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\parka\AppData\Local\Temp\mitm.sock
[2016-01-17 21:37:09.515] [DEBUG] proxy -   = will connect to socket "C:\Users\parka\AppData\Local\Temp\mitm.sock"
[2016-01-17 21:37:09.521] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\parka\AppData\Local\Temp\mitm.sock
[2016-01-17 21:37:19.463] [DEBUG] proxy -   = will connect to socket "C:\Users\parka\AppData\Local\Temp\mitm.sock"
[2016-01-17 21:37:19.468] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\parka\AppData\Local\Temp\mitm.sock
[2016-01-17 21:37:19.526] [DEBUG] proxy -   = will connect to socket "C:\Users\parka\AppData\Local\Temp\mitm.sock"
[2016-01-17 21:37:19.531] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\parka\AppData\Local\Temp\mitm.sock
[2016-01-17 21:38:19.471] [DEBUG] proxy -   = will connect to socket "C:\Users\parka\AppData\Local\Temp\mitm.sock"
[2016-01-17 21:38:19.476] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\parka\AppData\Local\Temp\mitm.sock
[2016-01-17 21:38:19.536] [DEBUG] proxy -   = will connect to socket "C:\Users\parka\AppData\Local\Temp\mitm.sock"
[2016-01-17 21:38:19.536] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\parka\AppData\Local\Temp\mitm.sock
[2016-01-17 21:38:19.571] [ERROR] proxy - > ERR: Error: read ECONNRESET
[2016-01-17 21:38:19.576] [ERROR] proxy - > ERR: Error: read ECONNRESET
[2016-01-17 21:38:19.576] [ERROR] proxy - > ERR: Error: read ECONNRESET
[2016-01-17 21:38:19.581] [ERROR] proxy - > ERR: Error: read ECONNRESET
[2016-01-17 21:38:19.581] [ERROR] proxy - > ERR: Error: read ECONNRESET
[2016-01-17 21:38:19.581] [ERROR] proxy - > ERR: Error: read ECONNRESET
```

**Comment by runk** [comment]
Apparently you're trying to run old version of module.

Try to upgrade to v0.4.1 which supports windows.

**Comment by Parka** [comment]
It happens with both v0.4.0 and v0.4.1.
Here is how I initialize it:
```
npm config set strict-ssl false & npm-proxy-cache -m 9000 -p 8080 -v -e -s "C:\Users\parka\node_cache" -l "C:\Users\parka\node_cache\log.log"
```

**Comment by runk** [comment]
Erm. This is weird. I've added a check for windows platforms in code which guides you yo use `--internal-port` configuration option when you're using windows.

What happens if you type `npm-proxy-cache --help`?

**Comment by runk** [comment]
I feel that you're still using older than 0.4 version somehow.. :/

**Comment by Parka** [comment]
`npm-proxy-cache --help`:
```
  Usage: npm-proxy-cache [options]

  Options:

    -h, --help                  output usage information
    -V, --version               output the version number
    -h, --host [name]           Hostname [localhost]
    -p, --port [number]         An integer argument [8080]
    -t, --ttl [seconds]         Cache lifetime in seconds [1800]
    -s, --storage [path]        Storage path
    -x, --proxy [address]       HTTP proxy to be used, e.g. http://user:pass@example.com:8888/
    -e, --expired               Use expired cache when npm registry unavailable
    -f, --friendly-names        Use actual file names instead of hashes in the cache
    -v, --verbose               Verbose mode
    -l, --log-path [path]       Log path
    -m, --internal-port [port]  HTTPs port to use for internal proxying "MITM" server (necessary for running on Windows systems)
```
`npm-proxy-cache --V`: `0.4.1`

**Comment by runk** [comment]
`--internal-port`

**Comment by tobiashochguertel** [comment]
@runk  I have the same problem.

I can't download via `npm-proxy-cache`, version 0.4.1. With an Powershell with administrator privileges there is no different behavior.
```
PS X:\> npm-proxy-cache.cmd --help

  Usage: npm-proxy-cache [options]

  Options:

    -h, --help                  output usage information
    -V, --version               output the version number
    -h, --host [name]           Hostname [localhost]
    -p, --port [number]         An integer argument [8080]
    -t, --ttl [seconds]         Cache lifetime in seconds [1800]
    -s, --storage [path]        Storage path
    -x, --proxy [address]       HTTP proxy to be used, e.g. http://user:pass@example.com:8888/
    -e, --expired               Use expired cache when npm registry unavailable
    -f, --friendly-names        Use actual file names instead of hashes in the cache
    -v, --verbose               Verbose mode
    -l, --log-path [path]       Log path
    -m, --internal-port [port]  HTTPs port to use for internal proxying "MITM" server (necessary for running on Windows systems)

PS X:\> npm-proxy-cache.cmd -V
0.4.1
PS X:\>
```


```
PS X:\internet-proxies> npm --proxy http://127.0.0.1:8000 --https-proxy http://127.0.0.1:8000 --strict-ssl false install bower --save
npm WARN package.json internet-proxies@1.0.0 No description
npm WARN package.json internet-proxies@1.0.0 No repository field.
npm WARN package.json internet-proxies@1.0.0 No README data
npm ERR! fetch failed https://registry.npmjs.org/bower/-/bower-1.7.7.tgz
npm WARN retry will retry, error on last attempt: Error: tunneling socket could not be established, statusCode=500
npm ERR! fetch failed https://registry.npmjs.org/bower/-/bower-1.7.7.tgz
npm WARN retry will retry, error on last attempt: Error: tunneling socket could not be established, statusCode=500
npm ERR! fetch failed https://registry.npmjs.org/bower/-/bower-1.7.7.tgz
```

**Comment by runk** [comment]
You should consider using `--internal-port` option

**Comment by tobiashochguertel** [comment]
@runk do you mean `--internal-port 8000` e.g. `npm-proxy-cache -h 127.0.0.1 -p 8000 --internal-port 8000` or `npm-proxy-cache -h 127.0.0.1 --internal-port 8000`?

**Comment by runk** [comment]
Either of them should work I think

**Comment by RemiArnaud** [comment]
same issue. Windows 10 if this makes a difference
get error when trying ```npm install```
```
npm -v
3.3.12

node -v
v4.2.6

npm-proxy-cache.cmd -V
0.4.1

.npmrc:
https-proxy = "http://127.0.0.1:8005/"
proxy = "http://127.0.0.1:8005/"
strict-ssl = false

npm-proxy-cache -e -v -h 127.0.0.1 -p 8005 --internal-port 8005
[ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\XXXX\AppData\Local\Temp\mitm.sock
```

**Comment by runk** [comment]
can you do `--internal-port` to be `8006`?

**Comment by runk** [comment]
It was fuckup from my end. Can you try version v0.4.2 please?

**Comment by RemiArnaud** [comment]
I don't think  v0.4.2 was released? https://github.com/runk/npm-proxy-cache/releases
you want me to test with the latest master in github, instead of ```npm install -g ``` ?

**Comment by runk** [comment]
https://www.npmjs.com/package/npm-proxy-cache

On 26 February 2016 at 10:33, Rémi Arnaud <notifications@github.com> wrote:

> I don't think v0.4.2 was released?
> https://github.com/runk/npm-proxy-cache/releases
> you want me to test with the latest master in github, instead of npm
> install -g ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/runk/npm-proxy-cache/issues/26#issuecomment-189034565>
> .
>



-- 
Cheers,
Dmitry

**Comment by DmitriyBerezin** [comment]
Have exactly the same problem "ERR: Error: connect ENOENT C:\Users\USERNAME\AppData\Local\Temp\mitm.sock" on my local machine with win 7 x64. I use npm-proxy-cache version 0.4.2. Here is the commands:
start cache service
```bash
npm-proxy-cache -e -h 127.0.0.1 -p 8005 -x http://proxy.company.ru:3128 -m
```
run npm install through proxy:
```bash
npm --proxy http://127.0.0.1:8005 --https-proxy http://127.0.0.1:8005 --strict-ssl false install
```

**Comment by runk** [comment]
That's weird. :/

I'll try to install Windows on my laptop and troubleshot the problem. Will keep you posted.

**Comment by runk** [comment]
@DmitriyBerezin can you do me a favour, open node repl in your terminal and type

```
process.platform
```

What's the value it has? `win32` or something else? I have a doubt that windows 10 has some funny value in it..

**Comment by DmitriyBerezin** [comment]
Yes, `win32`

**Comment by runk** [comment]
Okay, you forgot to specify the port number or internal server (mitm thingy). 

```
npm-proxy-cache -e -h 127.0.0.1 -p 8005 -x http://proxy.company.ru:3128 -m 8006
```

Should be the trick. I will make the error message more obvious.

**Comment by DmitriyBerezin** [comment]
I've tried your command with the specified port - the same error.

**Comment by runk** [comment]
Can you paste the output from proxy here please?

**Comment by DmitriyBerezin** [comment]
```bash
D:\Projects\Ouk2>npm-proxy-cache -e -h 127.0.0.1 -p 8005 -x http://proxy.frsd.ru:3128 -m 8006
->> 8006
[2016-07-29 15:42:22.748] [INFO] proxy - Listening on 127.0.0.1:8005 [20756]
[2016-07-29 16:03:25.529] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.535] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.536] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.536] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.537] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.537] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.537] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.538] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.538] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.538] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.539] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.539] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.539] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.540] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.540] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.540] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.542] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.543] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.543] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
```

**Comment by DmitriyBerezin** [comment]
```bash
D:\Projects\Ouk2>npm-proxy-cache -e -h 127.0.0.1 -p 8005 -x http://proxy.company.ru:3128 -m 8006
->> 8006
[2016-07-29 15:42:22.748] [INFO] proxy - Listening on 127.0.0.1:8005 [20756]
[2016-07-29 16:03:25.529] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.535] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.536] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.536] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.537] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.537] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.537] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.538] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.538] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.538] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.539] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.539] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.539] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.540] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.540] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.540] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.542] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.543] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
[2016-07-29 16:03:25.543] [ERROR] proxy - < ERR: Error: connect ENOENT C:\Users\username\AppData\Local\Temp\mitm.sock
```

**Comment by runk** [comment]
@DmitriyBerezin thanks! I just published the `v0.5.0`, can you try it?

**Comment by DmitriyBerezin** [comment]
@runk Thank you! Now works pretty well.
